head/src/main.rs

```diff
@@ -0,0 +1,138 @@
+use clap::{load_yaml, value_t, App, ArgMatches, ErrorKind};
+use std::fs::File;
+use std::io::{self, BufRead, BufReader, Read, Write};
+
+const DEFAULT_LINES_COUNT: usize = 10;
+const NEW_LINE: u8 = 0xA;
+
+fn main() {
+    let yaml = load_yaml!("head.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+    let flags = Flags::parse(&matches);
+    let input = Input::parse(&matches);
+
+    head(flags, input).unwrap_or_else(|e| {
+        eprintln!("head: {}", e);
+    });
+}
+
+/// We truncate the input at either some number of lines or bytes
+enum Flags {
+    LinesCount(usize),
+    BytesCount(usize),
+}
+
+impl Flags {
+    /// Parse arguments into a Flags enum
+    ///
+    /// This will exit the program early on invalid args
+    fn parse(matches: &ArgMatches) -> Self {
```

Comment:
Would be better if this function name was `from_matches`
In Rust is common to name the constructors have to receive arguments as "from_<kind/type/etc>"

Also change that in the `Input::parse()` as well